nailgun/entities.py

```diff
@@ -1359,11 +1360,19 @@ def __init__(self, server_config=None, **kwargs):
                 required=True,
             ),
         }
+        super(HostClasses, self).__init__(server_config, **kwargs)
         self._meta = {
-            'api_path': 'api/v2/hosts/:host_id/puppetclass_ids',
             'server_modes': ('sat'),
+            'api_path': '{0}/puppetclass_ids'.format(
+                # pylint:disable=no-member
+                self.host.path('self')
+            )
         }
-        super(HostClasses, self).__init__(server_config, **kwargs)
+
+    def read(self, entity=None, attrs=None, ignore=('hostgroup',)):
+        """Deal with redundant entity fields"""
+        if attrs is None:
+            attrs = self.read_json()
```

Comment:
If I can ask - why are we ignoring the `hostgroup` class? Is it because `hostgroup` is also an attribute of the `Host` entity, and therefore also listing the `hostgroup` attribute on `HostClass` is redundant? If that's the case, then we should be able to drop this method entirely (and possibly add something in to the docstring for this class explaining why the hostgroup attribute is ignored). NailGun just ignores extra parameters returned from the server. It only throws an exception when too few parameters are returned.